src/jit/codegenxarch.cpp

```diff
@@ -176,8 +176,44 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
 
     // Make sure that EAX is reported as live GC-ref so that any GC that kicks in while
     // executing GS cookie check will not collect the object pointed to by EAX.
-    if (!pushReg && (compiler->info.compRetType == TYP_REF))
-        gcInfo.gcRegGCrefSetCur |= RBM_INTRET;    
+    if (!pushReg)
+    {
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        if (varTypeIsStruct(compiler->info.compRetType))
+        {
+            if (compiler->info.retStructDesc.passedInRegisters)
+            {
+                assert(compiler->info.retStructDesc.eightByteCount > 0 &&
+                       compiler->info.retStructDesc.eightByteCount <= CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
+
+                for (unsigned i = 0; i < compiler->info.retStructDesc.eightByteCount; i++)
+                {
+                    var_types eightByteType = compiler->getEightByteType(compiler->info.retStructDesc, i);
+
+                    if (eightByteType == TYP_REF)
```

Comment:
This should be written this way:

if (varTypeIsGC(eightByteType) {
if (i==0) {  gcInfo.gcRegGCrefSetCur |= RBM_INTRET; }
else  {  gcInfo.gcRegGCrefSetCur |= RBM_INTRET_1; }
}
with appropriate whitespace.
You can also use the (expr) ? val1 : val2 
but I am not a fan of that, 
someone may want to put a breakpoint on the RBM_INTRET_1 
to find out what method hit this in mscorlib etc... 
and they will have to find another way of doing that.